mhs/mhs/common/state/tests/test_dynamo_persistence_adaoter.py

```diff
@@ -0,0 +1,145 @@
+import unittest
+from unittest.mock import MagicMock, patch, Mock
+
+from mhs.common.state import dynamo_persistence_adapter
+from mhs.common.state.dynamo_persistence_adapter import DynamoPersistenceAdapter, RecordCreationError, \
+    InvalidTableError, RecordRetrievalError, RecordDeletionError
+
+TEST_STATE_ARN = "TEST STATE ARN"
+TEST_ASYNC_ARN = "TEST ASYNC ARN"
+TEST_ASYNC_TABLE = "TEST ASYNC TABLE"
+TEST_STATE_TABLE = "TEST STATE TABLE"
+TEST_INVALID_TABLE = "TEST INVALID TABLE"
+TEST_DATA = "{\"test_attribute\": \"test_value\"}"
+TEST_KEY = "TEST_KEY"
+TEST_INVALID_KEY = "TEST INVALID KEY"
+TEST_ITEM = {"key": TEST_KEY, "data": TEST_DATA}
+TEST_GET_RESPONSE = {"Item": TEST_ITEM}
+TEST_GET_EMPTY_RESPONSE = {}
+TEST_DELETE_RESPONSE = {"Attributes": TEST_ITEM}
+TEST_ADD_RESPONSE = {"Attributes": TEST_ITEM}
+TEST_EXCEPTION = Exception()
+TEST_SIDE_EFFECT = Mock(side_effect=TEST_EXCEPTION)
+
+
+class TestDynamoPersistenceAdapter(unittest.TestCase):
```

Comment:
For consistency I think we're using the `Adaptor` spelling

Reply:
50:50 split, the original interface was called persistence_adapter...